waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingService.java

```diff
@@ -323,17 +338,20 @@ public PanopticOperationHandler getPanopticOperationHandler() {
 
       @Override
       public List<String> getAllDatabases() {
-        List<String> combined = new ArrayList<>();
-        try {
-          List<String> databases = primaryDatabasesCache.get(PRIMARY_KEY);
-          for (String database : databases) {
-            combined.add(primaryDatabaseMapping.transformOutboundDatabaseName(database));
+        List<String> federatedCombined = new ArrayList<>();
+        List<String> primaryCombined = new ArrayList<>();
+        for (Map.Entry<String, DatabaseMapping> entry : mappingsByDatabaseName.entrySet()) {
+          if (entry.getValue().equals(primaryDatabaseMapping)) {
+            primaryCombined.add(entry.getKey());
+          } else {
+            federatedCombined.add(entry.getKey());
           }
-          combined.addAll(mappingsByDatabaseName.keySet());
-        } catch (ExecutionException e) {
-          LOG.warn("Can't fetch databases: {}", e.getCause().getMessage());
         }
-        return combined;
+
+        Collections.sort(primaryCombined);
+        Collections.sort(federatedCombined);
+        primaryCombined.addAll(federatedCombined);
```

Comment:
why not sort the final list once before returning ? is this intentional ?

Reply:
If I sort it at the end, then I would get the order of databases like ['federated_db', 'primary_db'] and that's not an expected behaviour in the tests right now. I wasn't sure about this one, but if someone does manual mapped databases, would they be ok with combining the federated databases with the primary databases?

Reply:
@patduin may be able to confirm if there is a requirement to display databases  in this particular manner or they can all be in a sorted manner ?

Reply:
We should tamper with the order, whatever order we got back from the metastore should be preserved. Then we usually do, primary first, followed by federated metastores (as configured in the yml).